raiden/blockchain_events_handler.py

```diff
@@ -318,10 +324,14 @@ def handle_channel_settled(raiden: 'RaidenService', event: Event):
     to calculate the gain and potentially perform unlocks in case
     there is value to be gained.
     """
-    our_locksroot, partner_locksroot = get_onchain_locksroots(
-        raiden=raiden,
+    canonical_identifier = CanonicalIdentifier(
+        chain_identifier=CHAIN_ID_UNSPECIFIED,
         token_network_address=token_network_identifier,
         channel_identifier=channel_identifier,
+    )
+    our_locksroot, partner_locksroot = get_onchain_locksroots(
```

Comment:
This is buggy. Consider:

- There is a channel A-B
- Node B is stopped
- Node A closes and settles the channel
- `NO_STATE_QUERY_AFTER_BLOCKS` passes
- Node B is restarted

Then the call to `get_onchain_locksroots` will raise an exception.

@rakanalh @LefterisJP

Reply:
We had discussed this last week as a known problem and made this issue for it: https://github.com/raiden-network/raiden/issues/3778